API/RestMatch.API/RestMatch.API/Program.cs

```diff
@@ -14,6 +20,14 @@
 });
 
 builder.Services.AddControllers();
+
+builder.Services.AddAutoMapper(AppDomain.CurrentDomain.GetAssemblies());
+
+builder.Services.AddScoped<IRestaurantRepository, RestaurantRepository>();
```

Comment:
It's better to create separate extension methods for injecting repositories and services inside their layers and reuse here (see. injection of dbcontext)